component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPKeyGenerator.java

```diff
@@ -88,6 +88,7 @@
 
                 String issuer = TOTPUtil.getTOTPIssuerDisplayName(tenantDomain, context);
                 String displayUsername = TOTPUtil.getTOTPDisplayUsername(tenantAwareUsername);
+                displayUsername = getTOTPIssuerDisplayNameForFederatedUser(context, displayUsername);
```

Comment:
Check whether this is a federated user and call this. Otherwise this is confusing for local users. Noted that the check is inside the method. But, better to check the condition here.

Reply:
Here it is passing the display name to `getTOTPIssuerDisplayNameForFederatedUser` and if not federated the display name will  return back, Therefore no change affect for the local users.